homeassistant/components/nest/camera_sdm.py

```diff
@@ -94,21 +101,47 @@ async def stream_source(self):
         if CameraLiveStreamTrait.NAME not in self._device.traits:
             return None
         trait = self._device.traits[CameraLiveStreamTrait.NAME]
-        now = utcnow()
         if not self._stream:
-            logging.debug("Fetching stream url")
+            _LOGGER.debug("Fetching stream url")
             self._stream = await trait.generate_rtsp_stream()
-        elif self._stream.expires_at < now:
-            logging.debug("Stream expired, extending stream")
-            new_stream = await self._stream.extend_rtsp_stream()
-            self._stream = new_stream
+            self._schedule_stream_refresh()
+        if self._stream.expires_at < utcnow():
+            _LOGGER.warning("Stream already expired")
         return self._stream.rtsp_stream_url
 
+    def _schedule_stream_refresh(self):
+        """Schedules an alarm to refresh the stream url before expiration."""
+        _LOGGER.debug("New stream url expires at %s", self._stream.expires_at)
+        refresh_time = self._stream.expires_at - STREAM_EXPIRATION_BUFFER
+        # Schedule an alarm to extend the stream
+        self._stream_refresh_unsub = async_track_point_in_utc_time(
```

Comment:
Make sure you undo the old one

```suggestion
        if self._stream_refresh_unsub is not None:
            self._stream_refresh_unsub()

        self._stream_refresh_unsub = async_track_point_in_utc_time(
```